src/python/WMComponent/WorkflowUpdater/WorkflowUpdaterPoller.py

```diff
@@ -385,13 +387,17 @@ def adjustJSONSpec(self, puWflows, msPileupList, dest=None):
                         # {block1: [rses], block2: [rses], ...}
                         if checkChanges(jsonBlockLoc, msPUBlockLoc):
                             self.logger.info("Found differences between JSON and MSPileup content.")
-                            puNewJsonContent = updateBlockInfo(puJsonContent, msPUBlockLoc, self.logger)
+                            puNewJsonContent = updateBlockInfo(puJsonContent, msPUBlockLoc, self.dbsUrl, self.logger)
 
                             # we should update a tarball only once for each pileup name,
                             # therefore we add new entry to jdict with our pilupe conf file
-                            jdict[puConfJson] = puNewJsonContent
-                            self.logger.info("Mark %s to be updated in tarball %s with a fresh pileup content",
-                                             puConfJson, tarFile)
+                            if puNewJsonContent:
+                                # we update json file if we get new pileup content
+                                jdict[puConfJson] = puNewJsonContent
+                                self.logger.info("Mark %s to be updated in tarball %s with a fresh pileup content",
+                                                 puConfJson, tarFile)
+                            else:
+                                self.logger.info("updateBlockInfo did not return any results, will skipp update of pileup jsoon content")
```

Comment:
Please make this a warning log level. There is a few typos in this message as well.
In addition, I would suggest to add the pileup name in this message.

Reply:
this is fixed.